### PR TITLE
docs(tproxy): almost complete overhaul for transparent proxy docs

### PR DESCRIPTION
**Pull request by bartsmykla**
**Description of this PR is in progress and will be updated**

Links to preview for related pages:

Highly modified:
- [dp config / transparent proxy](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/production/dp-config/transparent-proxying/)

Very small changes:
- [networking / transparent proxying](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/networking/transparent-proxying/)

Content moved to guide:
- [installation](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/integrating-transparent-proxy-into-your-service-environment/)
- [upgrading](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/upgrading-transparent-proxy/)
- [excluding traffic from redirection](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/excluding-traffic-from-transparent-proxy-redirection/)

New guides:
- [modify tproxy config for kubernetes](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode/)
- [modify tproxy config for universal](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/customizing-transparent-proxy-configuration-in-universal-mode/)

New content:
- [tproxy config reference](https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/reference/transparent-proxy-configuration/)

---

- Rewrote transparent proxying docs from the dataplane configuration
- Moved docs related to installing, upgrading transparent proxy or excluding traffic to separate guides
- Created transparent proxy reference
- Created two guides related to customizing transparent proxy configuration
- Few small fixes in networking / transparent proxying page
- Created helper `inc` jekyll plugin which allows to conditionally increment some values which is helpful to render number of step when there is some conditional logic related to version included

Did you sign your commit? [Instructions](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md#sign-your-commits): 👍 

Have you read [Contributing guidelines](https://github.com/kumahq/.github/blob/main/CONTRIBUTING.md)?: 👍

### CODE REVIEW

File: app/_assets/entrypoints/application.js [hunk 1]
```diff
@@ -39,5 +39,5 @@ document.addEventListener('DOMContentLoaded', (event) => {
     placement: 'left',
     icon: '#'
   });
-  anchors.add('.doc h1, .doc h2, .doc h3, .doc h4, .doc h5, .doc h6');
+  anchors.add('.doc h1:not(.no-anchor), .doc h2:not(.no-anchor), .doc h3:not(.no-anchor), .doc h4:not(.no-anchor), .doc h5:not(.no-anchor), .doc h6:not(.no-anchor)');
```
Comment (slonka) [inline app/_assets/entrypoints/application.js:42]: why is this needed?

File: app/_data/docs_nav_kuma_2.2.x.yml [hunk 2]
```diff
@@ -553,6 +553,17 @@ items:
             url: "/policies/meshloadbalancingstrategy/#configuration"
           - text: Examples
             url: "/policies/meshloadbalancingstrategy/#examples"
+  - title: Guides
```
Comment (slonka) [inline app/_data/docs_nav_kuma_2.2.x.yml:556]: Are you sure everything here is applicable to `2.2.x`? also `2.5.x` goes out of support on `2024-11-15` so should we really be modifyting `2.2.x - 2.5.x`? WDYT?

File: app/_data/tproxy.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      excludePorts: &excludePortsInbound
+        path: runtime.kubernetes.injector.redirect.inbound.excludePorts
+        flag: --exclude-inbound-ports
+        env: KUMA_TRANSPARENT_PROXY_REDIRECT_INBOUND_EXCLUDE_PORTS
+        runtime:
+          path: sidecarTraffic.excludeInboundPorts
+          env: KUMA_RUNTIME_KUBERNETES_SIDECAR_TRAFFIC_EXCLUDE_INBOUND_PORTS
+      excludePortsForIPs: &excludePortsForIPsInbound
+        path: runtime.kubernetes.injector.redirect.inbound.excludePortsForIPs
+        flag: --exclude-inbound-ips
```
Comment (slonka) [inline app/_data/tproxy.yaml:13]: Is this correct? The path mentions `ports` and the flag does not 🤔.

File: app/_includes/snippets/tproxy/conf-field-table.html.liquid [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      <tr>
+        <td><strong>Environment Variable</strong></td>
+        <td><code>KUMA_RUNTIME_KUBERNETES_INJECTOR_{{ include.runtimeEnv }}</code></td>
+      </tr>
+    {% endif %}
+  {% endif %}
+  </tbody>
+</table>
+{%- endcapture -%}
+{{ snippet-tproxy-field-table | normalize_whitespace }}
```
Comment (slonka) [inline app/_includes/snippets/tproxy/conf-field-table.html.liquid:71]: nitpick newline:

```suggestion
{{ snippet-tproxy-field-table | normalize_whitespace }}

```

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+
+{% assign configuration-in-configmap = "Configuration in ConfigMap <sup>(experimental)</sup>" %}
+
+The default transparent proxy configuration works well for most scenarios, but there are cases where adjustments are needed. This guide explains the various methods available for modifying the configuration, along with their limitations and recommendations on when to use each one. While these methods can be used individually, they can also be combined when necessary.
+
+{% warning %}
+It’s best to stick to one method whenever possible. Using more than one can make things more complicated and harder to troubleshoot, as it may not be clear where each setting comes from. If you need to combine methods, check the [**Order of Precedence**](#order-of-precedence) to see what the final configuration will look like based on the priority of each setting.
+{% endwarning %}
+
+## Terminology Overview
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:22]: I think vale would like this:

```suggestion
## Terminology overview
```

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 6]
```diff
@@ -0,0 +1,434 @@
+---
+title: Customizing Transparent Proxy Configuration in Kubernetes Mode
+---
+
+{% assign docs = "/docs/" | append: page.version %}
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:5]: What are these assignments? I don't think I've seen them before 🤔

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+
+- **ConfigMap**: In this guide, **ConfigMap** refers to the Kubernetes [**ConfigMap**](https://kubernetes.io/docs/concepts/configuration/configmap/) resource. It’s used to store configuration data as key-value pairs that can be easily accessed by other resources in the cluster, such as Pods, Deployments, and StatefulSets.
+
+## Method {% inc method_number %}: Control Plane Runtime Configuration
+{:#{{ tproxy.ids.guides.customize-config.k8s.methods.runtime }}}
+
+The Control Plane Runtime Configuration is a section of {{ Kuma }}'s configuration that contains important settings for how the control plane operates, including options for the transparent proxy and other key components.
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:31]: Ok, so this {{ Kuma }} is actually {{ site.mesh_product_name }}.

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 8]
```diff
@@ -0,0 +1,434 @@
+---
+title: Customizing Transparent Proxy Configuration in Kubernetes Mode
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:2]: In general this feels more like a reference doc not a guide. I'd say a guide has some specific task in mind and explains how to do a thing.

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+- `runtime.kubernetes.injector.sidecarContainer.redirectPortInbound`
+- `runtime.kubernetes.injector.sidecarContainer.redirectPortOutbound`
+- `runtime.kubernetes.injector.sidecarContainer.uid`
+- `runtime.kubernetes.injector.builtinDNS.enabled`
+- `runtime.kubernetes.injector.builtinDNS.port`
+
+{% if_version gte:2.9.x %}
+{% capture behavior-difference-danger %}
+{% danger %}
+⚠️ This behavior differs from that of other settings. Ensure you're aware of this distinction and **never** attempt to modify these settings using the mentioned annotations.
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:86]: I don't understand this :(

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 10]
```diff
@@ -0,0 +1,434 @@
...
+- **`runtime.kubernetes.injector.sidecarContainer.redirectPortInbound` and `runtime.kubernetes.injector.sidecarContainer.redirectPortOutbound`**
+
+  Attempts to change these settings using `kuma.io/transparent-proxying-inbound-port` and `kuma.io/transparent-proxying-outbound-port` annotations, or by updating `redirect.inbound.port` and `redirect.outbound.port` in the ConfigMap, will trigger warnings in the control plane logs, but the values will be ignored.
+
+- **`runtime.kubernetes.injector.sidecarContainer.uid`**
+
+  Attempts to modify this setting using the `kuma.io/sidecar-uid` annotation or `kumaDPUser` in the ConfigMap will result in:
+
+  - A warning in control plane logs if the [{{ configuration-in-configmap }}](#{{ tproxy.ids.guides.customize-config.k8s.methods.configmap }}) feature is **enabled**, and the values will be ignored.
+  - Silent ignoring of the change without warnings if the feature is **disabled**.
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:101]: Nitpick (but maybe that's why vale is complaining)

```suggestion
  - Silently ignoring the change without warnings if the feature is **disabled**.
```

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 11]
```diff
@@ -0,0 +1,434 @@
...
+
+- `kuma.io/transparent-proxying-ebpf-bpf-fs-path`
+
+- `kuma.io/transparent-proxying-ebpf-cgroup-path`
+
+- `kuma.io/transparent-proxying-ebpf-tc-attach-iface`
+
+- `kuma.io/transparent-proxying-ebpf-instance-ip-env-var-name`
+
+- `kuma.io/transparent-proxying-ebpf-programs-source-path`
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:344]: this will eventually get out of date, should we just link to the sections in https://kuma.io/docs/2.9.x/reference/kubernetes-annotations ?

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md [hunk 12]
```diff
@@ -0,0 +1,434 @@
...
+The following {% if_version gte:2.9.x %}annotations differ{% endif_version %}{% if_version lte:2.8.x %}annotation differs{% endif_version %} from others mentioned earlier as they are related to the transparent proxy, but do not directly or indirectly configure specific individual settings.
+
+- `kuma.io/transparent-proxying`
+
+  This annotation is automatically applied to workloads with [injected sidecar containers]({{ docs }}/production/dp-config/dpp-on-kubernetes/#kubernetes-sidecar-containers). In Kubernetes environments, the transparent proxy is mandatory meaning you cannot disable it. Any attempt to explicitly disable it using this annotation will trigger a warning and have no effect. Therefore, the value of this annotation will always be set to `true`.
+
+{% if_version gte:2.9.x %}
+- `traffic.kuma.io/transparent-proxy-config`
+
+  {% comment %}This comment is necessary because `if_version` interferes with list formatting, causing the new line before the first paragraph in the first list item to be ignored. Adding this comment ensures an empty line is rendered.{% endcomment %}
```
Comment (slonka) [inline app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:388]: please create an issue for this

File: app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
+---
+title: Excluding Traffic from Transparent Proxy
```
Comment (slonka) [inline app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md:2]: Should we use some concrete piece of software that the users had problems with before and just show this:

1. the app does not work / start
2. we add excludes
3. the app works

? Would that be more concrete and relateable?

File: app/_src/guides/integrating-transparent-proxy-into-your-service-environment.md [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+
+## Prerequisites
+
+{%- capture control-plane-and-service-network-accessibility-warning %}
+{% warning %}
+For this guide, the Control Plane **should not** be deployed in the same environment as the transparent proxy. This is because the transparent proxy redirects **all** traffic (unless specifically excluded) to `kuma-dp`, including traffic meant for the Control Plane.
+
+While it is technically possible to manually exclude Control Plane traffic during setup, this is complex and easy to misconfigure, as it requires excluding **all** necessary ports and routes.
+
+In real deployments, you can work around it by running the Control Plane and other components in separate network namespaces using `systemd`, which keeps Control Plane traffic separate from `kuma-dp` and the service.
```
Comment (slonka) [inline app/_src/guides/integrating-transparent-proxy-into-your-service-environment.md:31]: I think this guide should show how to set up this in namespaces. We had multiple people asking about collisions etc.

File: app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+{% capture path-method-k8s-runtime %}{{ docs }}{{ tproxy.paths.guides.customize-config.k8s }}#{{ tproxy.ids.guides.customize-config.k8s.methods.runtime }}{% endcapture %}
+{% capture path-method-k8s-annotations %}{{ docs }}{{ tproxy.paths.guides.customize-config.k8s }}#{{ tproxy.ids.guides.customize-config.k8s.methods.annotations }}{% endcapture %}
+{% capture path-method-uni-env %}{{ docs }}{{ tproxy.paths.guides.customize-config.uni }}#{{ tproxy.ids.guides.customize-config.uni.methods.env }}{% endcapture %}
+{% capture path-method-uni-cli %}{{ docs }}{{ tproxy.paths.guides.customize-config.uni }}#{{ tproxy.ids.guides.customize-config.uni.methods.cli }}{% endcapture %}
+{% capture path-integrating-tproxy-title %}{{ docs }}{{ tproxy.paths.guides.integrate-tproxy }}#{{ tproxy.ids.guides.integrate-tproxy.title }}{% endcapture %}
+{% capture path-step-install-tproxy %}{{ docs }}{{ tproxy.paths.guides.integrate-tproxy }}#{{ tproxy.ids.guides.integrate-tproxy.steps.install-tproxy }}{% endcapture %}
+
+{% assign configuration-in-configmap = "Configuration in ConfigMap <sup>(experimental)</sup>" %}
```
Comment (slonka) [inline app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md:20]: Do we need all the options here? Should we just go through the most common path and link somewhere that ConfigMap is also an option?

File: app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+
+  - To generate a data plane proxy token (needed to authorize data plane proxies), refer to [Data Plane Proxy Token]({{ docs }}/production/secure-deployment/dp-auth/#data-plane-proxy-token) documentation.
+
+  - To prepare your service environment and start the data plane proxy, follow the [Integrating Transparent Proxy into Your Service Environment]({{ path-integrating-tproxy-title }}) guide up to [Step {% inc install-tproxy if_version="lte:2.4.x" init_value=5 %}: Install the Transparent Proxy]({{ path-step-install-tproxy }}).
+    {% endtab %}
+    {% endtabs %}
+    {% endcapture %}
+    {{ prerequisites-setup-kuma | indent }}
+
+## Exclude Traffic on Certain Ports
```
Comment (slonka) [inline app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md:80]: I appreciate that we have key information, terminology and prerequisites but it might be a bit much for a basic use case right? I think this guide can be really entry level and lightweight with other guides going into more details.

**Review by slonka — COMMENTED** [review]
It's too much for one take. I'll have another look. My general feeling is that `customizing-*` documents are more like reference docs not guides. Excluding / integrating / upgrading are really guides and a good move but I think we need to adjust the complexity of these.

File: app/_data/tproxy.yaml (hunk 3 above)
Comment (bartsmykla) [reply app/_data/tproxy.yaml:13]: it's correct, internally it's called excludePortsForIPs but this flag is configuring this field

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 5 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md]: already implemented this fix

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 6 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:5]: instead of doing `[link name](/docs/{{ page.version }}/...)` we can do now just `[link name]({{ docs }}/...` it's just a helper so it's easier to write links so you don't have to remember this earlier construct and can just do `{{ docs }}`

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 7 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md]: indeed, it's easier to write docs without having to remember `site.mesh_product_name` and can just do `{{ Kuma }}`. It's also easier to read the sentences in this way

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 8 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:2]: Where do you think it should be placed? Under the reference section in the sidebar?

File: app/_assets/entrypoints/application.js (hunk 1 above)
Comment (bartsmykla) [reply app/_assets/entrypoints/application.js:42]: it's to be able to mark some heading as one, which shouldn't be linked to, for example:
image
it's in tabs so we can't link to it and if we allow `anchorjs` to process it, it would not do so correctly:
image

File: app/_includes/snippets/tproxy/conf-field-table.html.liquid (hunk 4 above)
Comment (bartsmykla) [reply app/_includes/snippets/tproxy/conf-field-table.html.liquid:71]: addressed

File: app/_src/guides/integrating-transparent-proxy-into-your-service-environment.md (hunk 14 above)
Comment (bartsmykla) [reply app/_src/guides/integrating-transparent-proxy-into-your-service-environment.md:31]: Maybe as a follow up. I would argue this guide should stay like this, and maybe we could improve it or create another one in the future

File: app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md (hunk 16 above)
Comment (bartsmykla) [reply app/_src/guides/excluding-traffic-from-transparent-proxy-redirection.md]: I actually think these might be helpful. Maybe if we would introduce a functionality so we can fold some sections, and even fold by default it would be better?

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 10 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md]: addressed

File: app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md (hunk 11 above)
Comment (bartsmykla) [reply app/_src/guides/customizing-transparent-proxy-configuration-in-kubernetes-mode.md:352]: I don't think these will go out of date to be honest, and it's hard to reference some subsets and it would involve restructuring the reference page which I don't want to do as this PR is already quite big.

File: app/_data/docs_nav_kuma_2.2.x.yml (hunk 2 above)
Comment (bartsmykla) [reply app/_data/docs_nav_kuma_2.2.x.yml]: addressed

**Comment by slonka** [comment]
Preview rendered: 

https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/excluding-traffic-from-transparent-proxy-redirection/

https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/networking/transparent-proxying/

**Comment by bartsmykla** [comment]
> Preview rendered:
> 
> https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/guides/excluding-traffic-from-transparent-proxy-redirection/
> 
> https://deploy-preview-2049--kuma.netlify.app/docs/2.9.x/networking/transparent-proxying/

I have already linked to all previews in the PR description

**Comment by bartsmykla** [comment]
Closing this PR as I'm gonna split it into few after the discussion with @slonka and @Automaat
